<!--
   Copyright 2024 eve autonomy inc. All Rights Reserved.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
-->

<launch>
  <arg name="dio_ros_driver_config_path" default="$(find-pkg-share dio_ros_driver)/config/dio_ros_driver.params.yaml"/>
  <arg name="sleep_time_at_running_state" default="500"/>

  <node pkg="status_display_manager" exec="status_display_manager" name="status_display_manager" output="screen">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)" />
    <param from="$(var dio_ros_driver_config_path)"/>
    <param from="$(var sleep_time_at_running_state)"/>
    <remap from="display_status" to="/status_display_manager/state" />
    <remap from="emergency_switch_input" to="/diagnostics_err" />
    <remap from="turn_indicators_status" to="/vehicle/status/turn_indicators_status" />
  </node>
</launch>
